framework/tests/packets.rs

```diff
@@ -0,0 +1,318 @@
+        assert_eq!(eth.src.addr, MacAddress::new(0, 0, 0, 0, 0, 2).addr);
+        assert_eq!(eth.etype(), Some(EtherType::IPv6));
+    }
+
+    // Check IPv6 header
+    let v6pkt = epkt.parse_header::<Ipv6Header>();
+    {
+        let v6 = v6pkt.get_header();
+        let src = Ipv6Addr::from_str("2001:db8:85a3::1").unwrap();
+        let dst = Ipv6Addr::from_str("2001:db8:85a3::8a2e:0370:7334").unwrap();
+        assert_eq!(v6.version(), 6);
+        assert_eq!(v6.traffic_class(), 0);
+        assert_eq!(v6.flow_label(), 0);
+        assert_eq!(v6.payload_len(), 24);
+        assert_eq!(v6.next_header().unwrap(), NextHeader::Routing);
+        assert_eq!(v6.hop_limit(), 2);
+        assert_eq!(v6.peek_num_segments().unwrap(), 2);
+        assert_eq!(Ipv6Addr::from(v6.src()), src);
+        assert_eq!(Ipv6Addr::from(v6.dst()), dst);
+    }
+
+    // Check SRH
+    let srhpkt = v6pkt.parse_header::<SegmentRoutingHeader<Ipv6Header, U_>>();
+    {
+        let srh = srhpkt.get_header();
+        let seg0 = Ipv6Addr::from_str("2001:db8:85a3::8a2e:0370:7334").unwrap();
+        let seg1 = Ipv6Addr::from_str("2001:db8:85a3::8a2e:0370:7335").unwrap();
+        assert_eq!(srh.next_header().unwrap(), NextHeader::Udp);
+        assert_eq!(srh.hdr_ext_len(), 4);
+        assert_eq!(srh.routing_type(), 4);
+        assert_eq!(srh.segments_left(), 0);
+        assert_eq!(srh.last_entry(), 1);
+        assert_eq!(srh.protected(), false);
+        assert_eq!(srh.oam(), false);
+        assert_eq!(srh.alert(), false);
+        assert_eq!(srh.hmac(), false);
+        assert_eq!(srh.tag(), 0);
+        assert_eq!(srh.segments().len(), 2);
+        assert_eq!(srh.segments()[0], seg0);
+        assert_eq!(srh.segments()[1], seg1);
+    }
+}
+
+fn insert_static_srh_from_bytes() {
+    let packet_header = [
+        // --- Ethernet header ---
+        // Destination MAC
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x01,
+        // Source MAC
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x02,
+        // EtherType (IPv6)
+        0x86,
+        0xDD,
+        // --- IPv6 Header ---
+        // Version, Traffic Class, Flow Label
+        0x60,
+        0x00,
+        0x00,
+        0x00,
+        // Payload Length
+        0x00,
+        0x18,
+        // Next Header (UDP = 17)
+        0x11,
+        // Hop Limit
+        0x02,
+        // Source Address
+        0x20,
+        0x01,
+        0x0d,
+        0xb8,
+        0x85,
+        0xa3,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x01,
+        // Dest Address
+        0x20,
+        0x01,
+        0x0d,
+        0xb8,
+        0x85,
+        0xa3,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x8a,
+        0x2e,
+        0x03,
+        0x70,
+        0x73,
+        0x34,
+    ];
```

Comment:
For completeness, should we put a TCP/UDP packet at the end? Making sure that the inserted header doesn't mess up the rest of the payload? There's all that byte shifting going on to make room for the header not being exercised.

Reply:
@drunkirishcoder don't we just cover this in the main srv6-inject anyhow?

Reply:
True.